packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -789,17 +793,16 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         }
         this.lastSummaryHandle = context.ackHandle;
 
-        const { result, blobsShaToPathCacheLatest } = await PerformanceEvent.timedExecAsync(this.logger,
+        const { result, blobsShaToPathCacheLatest, treesPathToSummaryTreeCacheLatest } = await PerformanceEvent.timedExecAsync(this.logger,
             { eventName: "uploadSummaryWithContext" },
             async () => this.writeSummaryTree(this.lastSummaryHandle, summary));
         const id = result ? result.id : undefined;
         if (!result || !id) {
             throw new Error(`Failed to write summary tree`);
         }
-        if (blobsShaToPathCacheLatest) {
-            this.blobsShaToPathCache = blobsShaToPathCacheLatest;
-            this.blobsShaProposalHandle = id;
-        }
+        this.treesPathToSummaryTreeCache = treesPathToSummaryTreeCacheLatest;
```

Comment:
should these 3 assignments move directly into writeSummaryTree() ?
It feels like code would be simpler...

Reply:
Unless it's important to have an old value preserved until whole process is done (which looks like the case with this.treesPathToSummaryTreeCache - it's used by snapshot writing process)

Reply:
Yeah we want to assign the caches once the upload is done otherwise that can throw and then on retry we will use the wrong values. So its good to do after the whole process.